shnet_main.c

```diff
@@ -190,21 +192,22 @@ static int shnet_port_recv(struct shnet_port *port,
     recv->vec = recv->req.vec;
     recv->port = port;
 
-    if ((unsigned long)recv->req.vec[0].iov_base & (PAGE_SIZE -1)) {
+    nr_pages = recv->req.vec[0].iov_len / PAGE_SIZE + 1;
```

Comment:
What if the iov_len has PAGE_SIZE size? You would use more pages than needed, right?